### PR TITLE
Remove static pretrained maps from the library's internals

### PR DESCRIPTION
**Pull request by LysandreJik**

### CODE REVIEW

**Comment by LysandreJik** [comment]
Before this ungodly PR gets merged, I need to check that every checkpoint referenced here behaves the same once its pretrained map has been removed. 

I'll link the PRs open as a reuslt in this comment.

File: templates/adding_a_new_model/cookiecutter-template-{{cookiecutter.modelname}}/tokenization_{{cookiecutter.lowercase_modelname}}.py [hunk 1]
```diff
@@ -52,8 +43,6 @@ class {{cookiecutter.camelcase_modelname}}Tokenizer(BertTokenizer):
 
     vocab_files_names = VOCAB_FILES_NAMES
     pretrained_vocab_files_map = PRETRAINED_VOCAB_FILES_MAP
-    max_model_input_sizes = PRETRAINED_POSITIONAL_EMBEDDINGS_SIZES
-    pretrained_init_configuration = PRETRAINED_INIT_CONFIGURATION
 
 {%- elif cookiecutter.tokenizer_type == "Based on BART" %}
 from ...utils import logging
```
Comment (ArthurZucker) [inline templates/adding_a_new_model/cookiecutter-template-{{cookiecutter.modelname}}/tokenization_{{cookiecutter.lowercase_modelname}}.py:48]: The line with 
```
PRETRAINED_POSITIONAL_EMBEDDINGS_SIZES = {
    "{{cookiecutter.checkpoint_identifier}}": 1024,
}
```
should also be removed!

File: templates/adding_a_new_model/cookiecutter-template-{{cookiecutter.modelname}}/tokenization_fast_{{cookiecutter.lowercase_modelname}}.py [hunk 2]
```diff
@@ -53,8 +44,6 @@ class {{cookiecutter.camelcase_modelname}}TokenizerFast(BertTokenizerFast):
 
     vocab_files_names = VOCAB_FILES_NAMES
     pretrained_vocab_files_map = PRETRAINED_VOCAB_FILES_MAP
-    max_model_input_sizes = PRETRAINED_POSITIONAL_EMBEDDINGS_SIZES
-    pretrained_init_configuration = PRETRAINED_INIT_CONFIGURATION
     slow_tokenizer_class = {{cookiecutter.camelcase_modelname}}Tokenizer
 
 {%- elif cookiecutter.tokenizer_type == "Based on BART" %}
```
Comment (ArthurZucker) [inline templates/adding_a_new_model/cookiecutter-template-{{cookiecutter.modelname}}/tokenization_fast_{{cookiecutter.lowercase_modelname}}.py:49]: let's also cleanup for bart based

Comment (ArthurZucker) [reply templates/adding_a_new_model/cookiecutter-template-{{cookiecutter.modelname}}/tokenization_fast_{{cookiecutter.lowercase_modelname}}.py:49]: and `Standalone`

File: templates/adding_a_new_model/cookiecutter-template-{{cookiecutter.modelname}}/tokenization_{{cookiecutter.lowercase_modelname}}.py (hunk 1 above)
Comment (LysandreJik) [reply templates/adding_a_new_model/cookiecutter-template-{{cookiecutter.modelname}}/tokenization_{{cookiecutter.lowercase_modelname}}.py:48]: Correct! This needs to be adapted as well as the "new model addition" script that relies on these. Also there are 5k tests failing that I need to fix before I un-draft it :)

Comment (ArthurZucker) [reply templates/adding_a_new_model/cookiecutter-template-{{cookiecutter.modelname}}/tokenization_{{cookiecutter.lowercase_modelname}}.py:48]: Good luck!

**Review by julien-c — COMMENTED** [review]
casual G.O.A.T. = @LysandreJik

**Comment by LysandreJik** [comment]
Updated the base to move some of the refactor to that PR: https://github.com/huggingface/transformers/pull/29534

**Comment by LysandreJik** [comment]
Deprecate for 2 months and then simpler `DEL` a file and it's removed

**Comment by LysandreJik** [comment]
It should be good for review for the brave one @amyeroberts @ArthurZucker.

Failing tests are unrelated and also failing on `main`.

File: src/transformers/__init__.py [hunk 3]
```diff
@@ -5536,7 +5536,7 @@
         ProphetNetTokenizer,
     )
     from .models.pvt import PVT_PRETRAINED_CONFIG_ARCHIVE_MAP, PvtConfig
-    from .models.pvt_v2 import PVT_V2_PRETRAINED_CONFIG_ARCHIVE_MAP, PvtV2Config
+    from .models.pvt_v2 import PvtV2Config
     from .models.qdqbert import QDQBERT_PRETRAINED_CONFIG_ARCHIVE_MAP, QDQBertConfig
```
Comment (ArthurZucker) [inline src/transformers/__init__.py:5540]: let's remove all subsequent import to this as well

File: src/transformers/commands/add_new_model_like.py
Comment (ArthurZucker) [inline src/transformers/commands/add_new_model_like.py:1]: TODO for me, we need to make sure the the test add some pre-trained models to test!

File: src/transformers/models/deprecated/_archive_maps.py [hunk 4]
```diff
@@ -0,0 +1,2759 @@
...
+logger = logging.get_logger(__name__)
+
+
+class DeprecatedDict(dict):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    def __getitem__(self, item):
+        logger.warning(
+            "Archive maps are deprecated and will be removed in version v4.39.0 as they are no longer relevant. "
```
Comment (ArthurZucker) [inline src/transformers/models/deprecated/_archive_maps.py:29]:
```suggestion
            "Archive maps are deprecated and will be removed in version v4.40.0 as they are no longer relevant. "
```

File: src/transformers/models/deprecated/_archive_maps.py [hunk 5]
```diff
@@ -0,0 +1,2759 @@
...
+        return self[item]
+
+
+class DeprecatedList(list):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    def __getitem__(self, item):
+        logger.warning_once(
+            "Archive maps are deprecated and will be removed in version v4.39.0 as they are no longer relevant. "
```
Comment (ArthurZucker) [inline src/transformers/models/deprecated/_archive_maps.py:42]:
```suggestion
            "Archive maps are deprecated and will be removed in version v4.40.0 as they are no longer relevant. "
```

File: src/transformers/models/deprecated/retribert/tokenization_retribert.py [hunk 6]
```diff
@@ -112,8 +112,6 @@ class RetriBertTokenizer(PreTrainedTokenizer):
 
     vocab_files_names = VOCAB_FILES_NAMES
     pretrained_vocab_files_map = PRETRAINED_VOCAB_FILES_MAP
-    max_model_input_sizes = PRETRAINED_POSITIONAL_EMBEDDINGS_SIZES
-    pretrained_init_configuration = PRETRAINED_INIT_CONFIGURATION
```
Comment (ArthurZucker) [inline src/transformers/models/deprecated/retribert/tokenization_retribert.py:116]: the PRETRAINED_POSITIONAL_EMBEDDINGS_SIZES = {
    "yjernite/retribert-base-uncased": 512,
}
PRETRAINED_INIT_CONFIGURATION = {
    "yjernite/retribert-base-uncased": {"do_lower_case": True},
}
 were not removed

File: src/transformers/models/deprecated/retribert/tokenization_retribert_fast.py [hunk 7]
```diff
@@ -96,8 +96,6 @@ class RetriBertTokenizerFast(PreTrainedTokenizerFast):
 
     vocab_files_names = VOCAB_FILES_NAMES
     pretrained_vocab_files_map = PRETRAINED_VOCAB_FILES_MAP
-    max_model_input_sizes = PRETRAINED_POSITIONAL_EMBEDDINGS_SIZES
-    pretrained_init_configuration = PRETRAINED_INIT_CONFIGURATION
```
Comment (ArthurZucker) [inline src/transformers/models/deprecated/retribert/tokenization_retribert_fast.py:100]: same here

File: src/transformers/models/deprecated/tapex/tokenization_tapex.py [hunk 8]
```diff
@@ -265,8 +265,6 @@ class TapexTokenizer(PreTrainedTokenizer):
 
     vocab_files_names = VOCAB_FILES_NAMES
     pretrained_vocab_files_map = PRETRAINED_VOCAB_FILES_MAP
-    max_model_input_sizes = PRETRAINED_POSITIONAL_EMBEDDINGS_SIZES
-    pretrained_init_configuration = PRETRAINED_INIT_CONFIGURATION
```
Comment (ArthurZucker) [inline src/transformers/models/deprecated/tapex/tokenization_tapex.py:269]: same here

File: src/transformers/models/dpr/modeling_dpr.py [hunk 9]
```diff
@@ -39,18 +39,12 @@
...
-DPR_READER_PRETRAINED_MODEL_ARCHIVE_LIST = [
-    "facebook/dpr-reader-single-nq-base",
-    "facebook/dpr-reader-multiset-base",
-]
+
+from ..deprecated._archive_maps import (  # noqa: F401, E402
+    DPR_CONTEXT_ENCODER_PRETRAINED_MODEL_ARCHIVE_LIST,  # noqa: F401, E402
+    DPR_QUESTION_ENCODER_PRETRAINED_MODEL_ARCHIVE_LIST,  # noqa: F401, E402
+    DPR_READER_PRETRAINED_MODEL_ARCHIVE_LIST,  # noqa: F401, E402
+)
```
Comment (ArthurZucker) [inline src/transformers/models/dpr/modeling_dpr.py:47]:
```suggestion
from ..deprecated._archive_maps import DPR_CONTEXT_ENCODER_PRETRAINED_MODEL_ARCHIVE_LIST, DPR_QUESTION_ENCODER_PRETRAINED_MODEL_ARCHIVE_LIST, DPR_READER_PRETRAINED_MODEL_ARCHIVE_LIST #fmt skip # noqa: F401, E402

```
if that works

**Review by ArthurZucker — COMMENTED** [review]
First part

File: src/transformers/models/llama/tokenization_llama_fast.py [hunk 10]
```diff
@@ -33,16 +33,6 @@
...
-PRETRAINED_VOCAB_FILES_MAP = {
-    "vocab_file": {
-        "hf-internal-testing/llama-tokenizer": "https://huggingface.co/hf-internal-testing/llama-tokenizer/resolve/main/tokenizer.model",
-    },
-    "tokenizer_file": {
-        "hf-internal-testing/llama-tokenizer": "https://huggingface.co/hf-internal-testing/llama-tokenizer/resolve/main/tokenizer_config.json",
-    },
-}
-B_INST, E_INST = "[INST]", "[/INST]"
-B_SYS, E_SYS = "<<SYS>>\n", "\n<</SYS>>\n\n"
```
Comment (ArthurZucker) [inline src/transformers/models/llama/tokenization_llama_fast.py:45]: should not be deleted

File: templates/adding_a_new_model/cookiecutter-template-{{cookiecutter.modelname}}/to_replace_{{cookiecutter.lowercase_modelname}}.py [hunk 11]
```diff
@@ -122,16 +119,15 @@
...
 # End.
 
 # To replace in: "src/transformers/__init__.py"
 # Below: "        # PyTorch model imports" if generating PyTorch
 # Replace with:
 {% if cookiecutter.is_encoder_decoder_model == "False" %}
         from .models.{{cookiecutter.lowercase_modelname}} import (
-            {{cookiecutter.uppercase_modelname}}_PRETRAINED_MODEL_ARCHIVE_LIST,
-            {{cookiecutter.camelcase_modelname}}ForMaskedLM,
+            {{cookiecutter.uppercase_modelname}}            {{cookiecutter.camelcase_modelname}}ForMaskedLM,
```
Comment (ArthurZucker) [inline templates/adding_a_new_model/cookiecutter-template-{{cookiecutter.modelname}}/to_replace_{{cookiecutter.lowercase_modelname}}.py:130]: something is missing here

File: templates/adding_a_new_model/cookiecutter-template-{{cookiecutter.modelname}}/to_replace_{{cookiecutter.lowercase_modelname}}.py [hunk 12]
```diff
@@ -144,8 +140,7 @@
         )
 {% else %}
         from .models.{{cookiecutter.lowercase_modelname}} import (
-            {{cookiecutter.uppercase_modelname}}_PRETRAINED_MODEL_ARCHIVE_LIST,
-            {{cookiecutter.camelcase_modelname}}ForConditionalGeneration,
+            {{cookiecutter.uppercase_modelname}}            {{cookiecutter.camelcase_modelname}}ForConditionalGeneration,
```
Comment (ArthurZucker) [inline templates/adding_a_new_model/cookiecutter-template-{{cookiecutter.modelname}}/to_replace_{{cookiecutter.lowercase_modelname}}.py:143]: same here

**Review by ArthurZucker — APPROVED** [review]

File: src/transformers/models/dpr/modeling_dpr.py (hunk 9 above)
Comment (LysandreJik) [reply src/transformers/models/dpr/modeling_dpr.py:47]: prefer to keep it as is as I'll be looking for the suffix when removing these imports

**Pull request merged**